src/ADAL.PCL.Android/TokenCachePlugin.cs

```diff
@@ -16,22 +16,78 @@
 // limitations under the License.
 //----------------------------------------------------------------------
 
+using Android.Accounts;
+using Android.App;
+using Android.Content;
 using System;
 using System.Collections.Generic;
 using System.Linq;
 using System.Text;
+using System.Threading;
 using System.Threading.Tasks;
 
 namespace Microsoft.IdentityModel.Clients.ActiveDirectory
 {
     internal class TokenCachePlugin : ITokenCachePlugin
     {
+        private const string SharedPreferencesName = "ActiveDirectoryAuthenticationLibrary";
+        private const string SharedPreferencesKey = "cache";
+
+        public IntPtr Handle 
+        {
+            get { return IntPtr.Zero; }
+        }
+
         public void BeforeAccess(TokenCacheNotificationArgs args)
         {
+            if (args.TokenCache.Count > 0)
+            {
+                // We assume that the cache has not changed since last write
+                return;
+            }
+
+            try
+            {
+                ISharedPreferences preferences = Application.Context.GetSharedPreferences(SharedPreferencesName, FileCreationMode.Private);
+                string stateString = preferences.GetString(SharedPreferencesKey, null);
```

Comment:
This should be OK for the preview. I have opened an issue to follow up on encryption.